app/helpers/report_helper.rb

```diff
@@ -2,24 +2,16 @@
 require 'zip'
 
 module ReportHelper
-  def self.generate_report_as_job (org_id, account_filter, params)
-    @reports = ReportArchive.where(organization_id: org_id).all
-    @report = nil;
+  def self.generate_report_as_job (org_id, account_filter, params, report_id = nil)
+    @report = ReportArchive.find(report_id) if report_id
```

Comment:
check if there is already a report with the same params, if so, use that report so we don't have duplicate reports